src/internal_rules.jl

```diff
@@ -764,7 +764,7 @@ function EnzymeRules.augmented_primal(
     cache = if isa(A, Const)
         nothing
     else
-        dfact
+        (fact, dfact)
```

Comment:
fact is not always computed (e.g. if `EnzymeRules.needs_primal(config)` is false). Can you change fact to also be computed if `!isa(A, Const)` and also then have a `fact_returned = EnzymeRules.needs_primal(config) ? fact : nothing` which is the returned factors

Reply:
Also fact may be overwritten from forward to reverse so you should probably copy fact for the cache

Reply:
Thanks, need to look into this. This is not covered by any test utilities?

Reply:
Let me know if I got it right.